geode-core/src/main/java/org/apache/geode/distributed/internal/InternalLocator.java

```diff
@@ -651,20 +658,59 @@ public void onDisconnect(InternalDistributedSystem sys) {
   }
 
   private void startCache(DistributedSystem ds) {
-    InternalCache internalCache = GemFireCacheImpl.getInstance();
+    GemFireCacheImpl internalCache = GemFireCacheImpl.getInstance();
     if (internalCache == null) {
       logger.info("Creating cache for locator.");
-      this.myCache = (InternalCache) new CacheFactory(ds.getProperties()).create();
-      internalCache = this.myCache;
+      internalCache = (GemFireCacheImpl) new CacheFactory(ds.getProperties()).create();
+      this.myCache = internalCache;
     } else {
       logger.info("Using existing cache for locator.");
       ((InternalDistributedSystem) ds).handleResourceEvent(ResourceEvent.LOCATOR_START, this);
     }
     startJmxManagerLocationService(internalCache);
 
-    startSharedConfigurationService();
-    clusterManagementService = new LocatorClusterManagementService(locator.myCache,
-        locator.configurationPersistenceService);
+    startClusterManagementService();
+  }
+
+  private void startClusterManagementService() {
+    startConfigurationPersistenceService();
+
+    if (myCache == null) {
+      return;
+    }
+
+    clusterManagementService =
+        new LocatorClusterManagementService(myCache.getDistributionManager(),
+            locator.configurationPersistenceService);
+
+    // start management rest service
+    AgentUtil agentUtil = new AgentUtil(GemFireVersion.getGemFireVersion());
+    // Find the V2 Management rest WAR file
+    final String gemfireManagementWar = agentUtil.findWarLocation("geode-web-management");
+    if (gemfireManagementWar == null) {
+      logger.info(
+          "Unable to find GemFire V2 Management REST API WAR file; the Management REST Interface for Geode will not be accessible.");
+      return;
+    }
+
+    ManagementAgent managementAgent =
+        ((SystemManagementService) SystemManagementService.getExistingManagementService(myCache))
+            .getManagementAgent();
+
+    if (managementAgent == null) {
+      logger.info(
+          "management service needs to be started for ClusterManagementService to be running.");
+      return;
+    }
+
+    Object[] securityServiceAttr = new Object[] {JettyHelper.SECURITY_SERVICE_SERVLET_CONTEXT_PARAM,
+        myCache.getSecurityService()};
+    Object[] cmServiceAttr = new Object[] {JettyHelper.CLUSTER_MANAGEMENT_SERVICE_CONTEXT_PARAM,
```

Comment:
Seems more obvious to just use a `Properties` object here. I don't know why we made the underlying methods use an array of (implicit) `Object[2]` arguments.

Reply:
just to avoid having to create a Property and manipulate it based on what webapp it is.  one webapp only requires security service, another requires security service and sslConfig, and another requires securityService and managementService. So instead of 
```
Properties properties = new Properties();
properties.put("securityService", securityService);
addWebapp(context1, war1, properties);

properties.put("sslConfig", sslConfig);
addWebapp(context2, war2, properties);
```

I could do
```
addWebapp(context1, war1, new Object[]{"securityService", securityService});
addWebapp(context2, war2, new Object[]{"sslConfig", sslConfig}, new Object[]{"securityService", securityService});
```

Reply:
Looking at the code that calls the `addWebapp` method it's fairly obvious what's happening, however the actual method signature of `addWebapp` completely loses any semantic notion that it's being called with tuples of key/value pairs. Sorry if it seems pedantic, but please would you change it. :)